spec.md

```diff
@@ -592,6 +594,204 @@ After implementing the first endpoint `GET /v2/catalog` documented
+          "username": "mysqluser",
+          "password": "pass",
+          "host": "mysqlhost",
+          "port": 3306,
+          "database": "dbname"
+        }
+      },
+      {
+        "name": "internal-store",
+        "service_id": "90c7c453-c6a3-47ef-8284-5a2c00f4f18b",
+        "service_name": "the-database",
+        "plan_id": "d9301d66-7621-4526-9b97-63e93ddc2d13",
+        "plan_name": "big",
+        "instance_id": "0c7ec11f-977b-4027-a4d2-3dd81a9fbeb6"
+      }
+    ]
+  }
+}
+```
+
+### Dependencies
+
+| Field | Type | Description |
+| --- | --- | --- |
+| open\_dependencies\_parameter | string | A parameter name used for an open list of dependencies. |
+| services | [array-of-service-dependency-objects](#service-dependencies) | A list of service dependencies. |
+| instances | [array-of-instance-dependency-objects](#instance-dependencies) | A list of instance dependencies. |
+
+\* Fields with an asterisk are REQUIRED.
+
+### Service Dependencies
+
+| Field | Type | Description |
+| --- | --- | --- |
+| parameter* | string | Name of the parameter. MUST be a non-empty string and MUST be unique within all dependency parameters. |
+| description* | string | A short description of the dependency. MUST be a non-empty string. |
+| optional  | boolean | Whether the dependency is optional or not. Defaults to false. |
+| service_id | string | Service ID of the instance. MUST be a non-empty string. MUST NOT be used in conjunction with the fields `plan_ids` and `service_name`. |
+| plan_ids | array-of-strings | An array of plan IDs. The instance must match one of those plans. MUST be a non-empty array of non-empty strings. MUST NOT be used in conjunction with the fields `service_id` and `service_name`. |
+| service_name | string | Service name of the instance. MUST be a non-empty string. MUST NOT be used in conjunction with the fields `service_id` and `plan_ids`. |
+| plan_names | array-of-strings | An array of plan names. The instance must match one of those plans. MUST be a non-empty array of non-empty strings. REQUIRES the field `service_name`. |
+
+\* Fields with an asterisk are REQUIRED.
+
+### Instance Dependencies
+
+| Field | Type | Description |
+| --- | --- | --- |
+| parameter* | string | Name of the parameter. MUST be a non-empty string and MUST be unique within all dependency parameters. |
+| description* | string | A short description of the dependency. MUST be a non-empty string. |
+| optional  | boolean | Whether the dependency is optional or not. Defaults to false. |
+| instance\_id\_parameter | string | The name of the parameter that contains the service instance ID of an existing service. MUST be a non-empty string. MUST NOT be used in conjunction with the field `alias_parameter`. |
+| alias_parameter | string | The name of the parameter that contains the alias of an existing service. MUST be a non-empty string. MUST NOT be used in conjunction with the field `instance_id_parameter`. |
```

Comment:
also field i think. Also I think it should be a path.

I don't understand why there is a spec defined location for alias in provision but it does not end up in a first class location for the instance.

Reply:
I am not sure I see the value in alias. I suppose it's goal is to allow you to depend on an instance called `my_sql_database` but this has no mechanism for allowing the platform to know which of the currently available service instances might be able to fit into that slot, because I can name my services anything I want...

Or is this intended to be a dynamic field that is back-channeled into the catalog and the platform must assume it needs to do a catalog fetch after every provision?